clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/isolated/IsolatedIngestionServer.java

```diff
@@ -475,19 +462,50 @@ public boolean isResourceSubscribed(String topicName, int partition) {
   }
 
   public void maybeSubscribeNewResource(String topicName, int partition) {
-    topicPartitionSubscriptionMap.computeIfAbsent(topicName, s -> new VeniceConcurrentHashMap<>())
+    getTopicPartitionSubscriptionMap().computeIfAbsent(topicName, s -> new VeniceConcurrentHashMap<>())
         .computeIfAbsent(partition, p -> new AtomicBoolean(true));
   }
 
   Map<String, Map<Integer, AtomicBoolean>> getTopicPartitionSubscriptionMap() {
     return topicPartitionSubscriptionMap;
   }
 
+  ExecutorService getStatusReportingExecutor() {
+    return statusReportingExecutor;
+  }
+
+  IsolatedIngestionRequestClient getReportClient() {
+    return reportClient;
+  }
+
+  void stopConsumptionAndReport(IngestionTaskReport report) {
+    String topicName = report.topicName.toString();
+    int partitionId = report.partitionId;
+    Future<?> executionFuture = submitStopConsumptionAndCloseStorageTask(report);
+    getStatusReportingExecutor().execute(() -> {
+      LOGGER.info("DEBUGGING 1");
+      try {
+        executionFuture.get();
+      } catch (ExecutionException | InterruptedException e) {
+        LOGGER.warn(
+            "Encounter exception when trying to stop consumption and close storage for {} of topic: {}",
+            partitionId,
+            topicName);
+      }
+      LOGGER.info("DEBUGGING 2");
+      if (getReportClient().reportIngestionStatus(report)) {
+        LOGGER.info("DEBUGGING 3");
+        setResourceToBeUnsubscribed(topicName, partitionId);
```

Comment:
I understand that after making this change, server will be robust to the edge case you mentioned. However, let's put some thoughts on this: what will happen to the system if report client fails to report status, and thus isolated process doesn't unsubscribe from the partition? Will the main process become stuck somewhere, or will this action be retried somewhere else?

Reply:
synced offline, decided to add some detailed log for the original exception for now.